src/map.rs

```diff
@@ -66,35 +66,45 @@ impl<'a, R: 'a + Read> de::MapVisitor for MapVisitor<'a, R> {
         }
     }
 
-    fn size_hint(&self) -> (usize, Option<usize>) {
-        (self.attrs.size_hint().0, None)
+    fn size_hint(&self) -> Option<usize> {
+        self.attrs.size_hint().1
     }
 }
 
 struct AttrValueDeserializer(String);
 
-impl de::Deserializer for AttrValueDeserializer {
+impl<'de> de::Deserializer<'de> for AttrValueDeserializer {
     type Error = Error;
 
-    fn deserialize<V: Visitor>(self, visitor: V) -> VResult<V> {
+    fn deserialize_any<V: Visitor<'de>>(self, visitor: V) -> VResult<V::Value> {
         visitor.visit_string(self.0)
     }
 
-    fn deserialize_enum<V: Visitor>(self, _name: &str, _variants: &'static [&'static str], visitor: V) -> VResult<V> {
+    fn deserialize_u8<V: de::Visitor<'de>>(self, visitor: V) -> VResult<V::Value> {
+        let u = self.0.parse().map_err(Error::ParseIntError)?;
```

Comment:
Because some of your migrated tests are failing with descriptive text indicating they expect u8/i32 instead of strings and no test failures indicate similar for i64/i64.

IIRC the new methods in map.rs are needed to fix tests that are currently failing in master,  and may not be strictly necessary as part of the serde 1.0 update.